File: board_dsl/src/main/java/com/dooho/board/entity/LikyEntity.java
package com.dooho.board.entity;

import com.dooho.board.dto.board.LikyDto;
import jakarta.persistence.*;
import lombok.*;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity(name = "Liky")
@Table(name = "Liky")
public class LikyEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer likeId;
    private Integer boardNumber;
    private String userEmail;
    private String likeUserProfile;
    private String likeUserNickname;

    public LikyEntity(LikyDto dto) {
        this.likeId = dto.getLikeId();
        this.boardNumber = dto.getBoardNumber();
        this.userEmail = dto.getUserEmail();
        this.likeUserProfile = dto.getLikeUserProfile();
        this.likeUserNickname = dto.getLikeUserNickname();
    }
}

File: board_spring/src/main/java/com/dooho/board/entity/LikyEntity.kt
package com.dooho.board.entity

import com.dooho.board.dto.board.LikyDto
import jakarta.persistence.*



@Entity(name = "Liky")
@Table(name = "Liky")
data class LikyEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val likeId: Int? = null,
    val boardNumber: Int? = null,
    val userEmail: String? = null,
    val likeUserProfile: String? = null,
    val likeUserNickname: String? = null
) {
    constructor(dto: LikyDto?) : this(
        dto?.likeId,
        dto?.boardNumber,
        dto?.userEmail,
        dto?.likeUserProfile,
        dto?.likeUserNickname
    )
}
